### ISSUE TITLE
Numpy inconsistency

### ISSUE THREAD

**Issue by skuschel** [skuschel/postpic#221]
On the current master (v0.4+54.g66bc0f4), there is an inconsistency how postpic handles fields, when a dimension gets lost due to indexing:
```python
import numpy as np
m = np.random.random((10,12,15))
print(m[:, 4, :].shape)
```
prints `(10, 15)`.

```python
import postpic as pp
f = pp.Field(m)
print(f[:, 4, :].shape)
```
prints `(10, 1, 15)`

This is very helpful, when the data is exported to vtk: The field remembers its 3D position, such that paraview will display it at the correct location. However, it makes the ouput incompatible with `imshow` from matplotlib. In order to plot the data, an additional `squeze()` is required. Quite harmless, but also quite unexpected. Is that a bug or a feature?

**Comment by Ablinne** [comment]
I remember that we have discussed this already a long time ago and decided to keep it as ist was for the time being. But recently I watched myself having to do the explicit `squeeze()` frequently and wished we had changed that back then.
I vote to change the behaviour and get closer to `numpy`.

**Comment by Ablinne** [comment]
Doing `squeeze()` automatically in `__array__` solves the `imshow` problem, but it causes `field.shape != np.array(field).shape` which is also inconsistent.
Back when I worked on the `ufuncs` i noticed that the reducing `ufuncs` like `sum` have a `keepdims` argument which specifies whether a length-1 dimension will be kept or squeezed.
Maybe we could have this as a user-defined setting whether field[:, 0.0, :] will keep the length-1 dimension or not. Maybe a boolean member `Field.keepdims` that defaults to `False` to make the behaviour closer to numpy but makes it easy to return to the old behaviour?

**Comment by skuschel** [comment]
1) The entire thing is mostly interesting for the 3D export. In addition it would be nice to know for plotting that we are looking at the y=4um plane or similar. Therefore in the moment a length 1 dimension gets squeezed, we should add a note in the resulting field. I propose to add a dictionary `parameters` as a new field property, which can be used to store {'y':4e-6} or maybe even other arbitrary parameters, like {'a0:4, 'chirp':200e-30}. I think of a dictionary as the parameter list can be used definded and extended.
2) I realized that
```python
import numpy as np
m = np.random.random((10,12,15))
print(m[:, 4:5, :].shape)
```
returns `(10,1,15)`, which means that the user can this syntax to preserve the dimension for any reason and another attribute like `keepdims` is unnecessary. 

That would be fully numpy  compatible (2) and we would not loose information on the way (1).

**Comment by Ablinne** [comment]
1. Adding functionality like this should be well thought out, not just a qick & dirty fix to making sure information is not lost. If this is to be done, we should first explore more use cases and how it could be used/incorporated throughout the code base.

2. For integer indices this is easy, while it may not be so easy for axis-value based indexing. It is not a-priori easy to make sure that exactly length-1 along a particular axis is returned. Another Idea would be to create a class KeepDim, that could be used like this:

```
field.shape == (x,y,z)
field[:, 0.0, :].shape == (x, z)
field[:, KeepDim(0.0), :].shape == (x,1,z)
```

**Comment by Ablinne** [comment]
fixed via #230
